src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -137,9 +145,40 @@ static Controller fromOptions(Environment env,
         int scalingParameter = options.containsKey(STARTING_SCALING_PARAMETER) ? Integer.parseInt(options.get(STARTING_SCALING_PARAMETER)) : DEFAULT_STARTING_SCALING_PARAMETER;
         int[] scalingParameters = new int[32];
         int[] previousScalingParameters = new int[32];
-        //set the scaling parameter for each level to the starting scaling parameter or default scaling parameter
-        Arrays.fill(scalingParameters, scalingParameter);
-        Arrays.fill(previousScalingParameters, scalingParameter);
+        //check to see if we have scaling parameters stored and read those if able, otherwise use starting values
+        File f = new File(SCALING_PARAMETER_FILE_PATH, SCALING_PARAMETER_FILE_NAME);
+        if (f.length() != 0 && f.isReadable())
+        {
+            try
+            {
+                BufferedReader br = new BufferedReader(new FileReader(f));
+                String scalingParametersString = br.readLine();
+                String[] scalingParametersStrings = scalingParametersString.split(",");
```

Comment:
I made a comment earlier but maybe you missed it: https://github.com/ethan-brown2022/cassandra/commit/f855f540eafdac0ebd24983e640e959a2c5f1400#r113073332

We need to consider what happens to these files if we ever want to change the format.